### PR TITLE
Add text-field and tooltip

### PR DESCRIPTION
**Pull request by haoxins**

### CODE REVIEW

File: src/text-field.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      'mdl-js-button': button,
+      'mdl-button--icon': icon,
+      'mdl-textfield__expandable-holder': expandableHolder
+    }, className);
+
+    const inputClasses = classnames(inputBaseClasses);
+
+    const labelClasses = classnames(labelBaseClasses);
+
+    let textField, labelField = '', errorField = '';
```
Comment (phated) [inline src/text-field.js]: please define each with a separate `let`.  Also, I think they should all be defaulted to undefined so you don't end up with extra `<span>`s in the react output

File: src/text-field.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        {labelField}
+        {errorField}
+      </div>
+    );
+  }
+}
+
+Textfield.propTypes = {
+  className: React.PropTypes.string,
+  id: React.PropTypes.string.isRequired,
```
Comment (phated) [inline src/text-field.js]: could `id` be uniquely generated if not given?  seems weird to be required just to hook the label to the input

**Comment by phated** [comment]
Just a couple of things.  Great work, thanks!

**Comment by haoxins** [comment]
done! :smile:

**Comment by phated** [comment]
Excellent! Thanks! Published as 0.4.0

**Pull request merged**